src/jswrap_storage.c

```diff
@@ -927,3 +927,18 @@ void jswrap_storagefile_erase(JsVar *f) {
   jsvObjectSetChildAndUnLock(f,"addr",jsvNewFromInteger(0));
   jsvObjectSetChildAndUnLock(f,"mode",jsvNewFromInteger(0));
 }
+
+
+/*JSON{
+  "type" : "method",
+  "class" : "StorageFile",
+  "name" : "pipe",
+  "ifndef" : "SAVE_ON_FLASH",
+  "generate" : "jswrap_pipe",
+  "params" : [
+    ["destination","JsVar","The destination file/stream that will receive content from the source."],
+    ["options","JsVar",["[optional] An object `{ chunkSize : int=32, end : bool=true, complete : function }`","chunkSize : The amount of data to pipe from source to destination at a time","complete : a function to call when the pipe activity is complete","end : call the 'end' function on the destination when the source is finished"]]
+  ]
```

Comment:
We could type that argument if you like:
```suggestion
    "typescript": "pipe(dest: any, options?: { chunkSize?: number, end?: boolean, complete?: () => void }): void"
  ]
```

Reply:
We should probably do that in all the places it's defined: https://github.com/espruino/Espruino/search?q=jswrap_pipe

Reply:
Maybe it makes sense for you to do another PR for that @bobrippling ?

Reply:
Sure, it would be my pleasure